src/providers/whatsapp/whatsapp.service.ts

```diff
@@ -56,80 +56,112 @@ export class WhatsappService {
+    } catch (error) {
+      return { error: 'Conecte o whatsapp' };
+    }
   }
 
   async sendVideoMessage(data: SendMessageVideoAsGifDto) {
     const { to, path, filename, subtitle } = data;
     const formattedNumber = this.handleNumber(to);
-    const videoMessage = await this.client.sendVideoAsGif(
-      formattedNumber,
-      path,
-      filename,
-      subtitle,
-    );
-
-    return videoMessage;
+    try {
+      const videoMessage = await this.client.sendVideoAsGif(
+        formattedNumber,
+        path,
+        filename,
+        subtitle,
+      );
+
+      return videoMessage;
+    } catch (error) {
+      return { error: 'Conecte o whatsapp' };
+    }
   }
 
   async sendImageMessage(data: SendMessageImageDto) {
     const { to, path, filename } = data;
     const formattedNumber = this.handleNumber(to);
-    const imageMessage = await this.client.sendImage(
-      formattedNumber,
-      path,
-      filename,
-    );
-
-    return imageMessage;
+    try {
+      const imageMessage = await this.client.sendImage(
+        formattedNumber,
+        path,
+        filename,
+      );
+
+      return imageMessage;
+    } catch (error) {
+      return { error: 'Conecte o whatsapp' };
+    }
   }
 
   async sendVoiceMessage(data: SendMessageVoiceDto) {
     const { to, path } = data;
     const formattedNumber = this.handleNumber(to);
-    const voiceMessage = await this.client.sendVoice(formattedNumber, path);
+    try {
+      const voiceMessage = await this.client.sendVoice(formattedNumber, path);
 
-    return voiceMessage;
+      return voiceMessage;
+    } catch (error) {
+      return { error: 'Conecte o whatsapp' };
+    }
   }
 
   async sendFileDocument(data: SendFileDocumentDto) {
     const { to, path, filename, subtitle } = data;
     const formattedNumber = this.handleNumber(to);
-    const fileDocument = await this.client.sendFile(
-      formattedNumber,
-      path,
-      filename,
-      subtitle,
-    );
-
-    return fileDocument;
+    try {
+      const fileDocument = await this.client.sendFile(
+        formattedNumber,
+        path,
+        filename,
+        subtitle,
+      );
+
+      return fileDocument;
+    } catch (error) {
+      return { error: 'Conecte o whatsapp' };
+    }
   }
   async getContactList() {
-    const contacts = await this.client.getAllContacts();
-    return contacts;
+    try {
+      const contacts = await this.client.getAllContacts();
+      return contacts;
+    } catch (error) {
+      return { error: 'Conecte o whatsapp' };
+    }
   }
 
   async getContact(phoneNumber: string) {
-    const formattedNumber = this.handleNumber(phoneNumber);
-    const contact = await this.client.getContact(formattedNumber);
-    return contact;
+    try {
+      const formattedNumber = this.handleNumber(phoneNumber);
+      const contact = await this.client.getContact(formattedNumber);
+      return contact;
+    } catch (error) {
+      return { error: 'Conecte o whatsapp' };
+    }
```

Comment:
tu pode melhorar isso, tu ta repetindo esse try/catch multipas vezes. Pensa um pouquinho ai, se tu travar, me avisa

Reply:
eu posso fazer com .then, fica melhor?

Reply:
nao, tem outra maneira melhor

Reply:
fiz desse jeito:

async sendVoiceMessage(data: SendMessageVoiceDto) {
    const { to, path } = data;
    const formattedNumber = this.handleNumber(to);

    const voiceMessage = await this.handleError(
      'sendVoiceMessage',
      formattedNumber,
      path,
    );

    return voiceMessage;
  }

private async handleError(
    methodName: string,
    formattedNumber: any,
    path?: string,
  ) {
    try {
      switch (methodName) {
        case 'sendVoiceMessage':
          const contact = await this.client.sendVoice(formattedNumber, path);
          return contact;
          break;

        default:
          break;
      }
    } catch (error) {
      return { error: 'Whatsapp desconectado' };
    }
  }

Reply:
faz algo parecido com isso aqui:

```suggestion
    private checaWhatsapp() {
         if(!this.client) throw new Error('WhatsApp desconectado')
  }
```

Reply:
mas sem apagar o codigo ali kkkk, é só um exemplo

Reply:
ah blz